src/core/file.cc

```diff
@@ -1048,20 +1048,74 @@ static internal::alignments xfs_alignments(const dioattr& da, unsigned block_siz
     return ret;
 }
 
+// Query block device alignment properties using ioctl and statx
+static
+future<internal::alignments>
```

Comment:
This function is not really a "coroutine", it only returns ready futures all the time